pkg/accounting/accounting.go

```diff
@@ -380,3 +452,49 @@ func (a *Accounting) NotifyPayment(peer swarm.Address, amount uint64) error {
 
 	return nil
 }
+
+// subtractI64mU64 is a helper function for safe subtraction of Int64 - Uint64
+// It checks for
+//   - overflow safety in conversion of uint64 to int64
+//   - safety of the arithmetic
+//   - whether ( -1 * result ) is still Int64, as MinInt64 in absolute sense is 1 larger than MaxInt64
+// If result is MinInt64, we also return overflow error, for two reasons:
+//   - in some cases we are going to use -1 * result in the following operations, which is secured by this check
+//   - we also do not want to possibly store this value as balance, even if ( -1 * result ) is not used immediately afterwards, because it could
+//		disable settleing for this amount as the value would create overflow
+
+func subtractI64mU64(base int64, subtracted uint64) (result int64, err error) {
+	result = base - int64(subtracted)
```

Comment:
i find the order of operations here a bit unusual. it would be clearer to check whether `subtracted` overflows `int64` first